frontend/packages/noobaa-storage-plugin/src/components/capacity-card/capacity-card-body.tsx

```diff
@@ -1,72 +1,52 @@
 import * as React from 'react';
-import * as _ from 'lodash';
 import { ChartDonut, ChartThemeVariant, ChartThemeColor } from '@patternfly/react-charts';
 import { ChartPieIcon } from '@patternfly/react-icons';
 import { DataPoint } from '@console/internal/components/graphs';
 import { EmptyState, EmptyStateVariant, EmptyStateIcon, Title } from '@patternfly/react-core';
 import { LoadingInline, humanizeBinaryBytesWithoutB } from '@console/internal/components/utils';
-
-type MetricDataType = {
-  x: string;
-  y: number;
-};
+import './capacity-card-body.scss';
 
 type LegendDataType = {
   name: string;
 };
 
 // as per the noobaa spec,
-// a. omit 'Others' label
-// b. sort it in descending order
-// c. collect the first 'N' items
-// d. rest of the miscs data should be aggregated
-//    into 'Others' label,
-// and separating Data and Legend array from DataPoint
+// a. sort it in descending order and pick the top 6 items (excluding 'Others' label)
+// b. append the 'Others' label
+// PS: 'metricData' argument passed to this function will be in sorted form and
+//     with 'Others' label appended to it
 const getMetricDataAndLegend = (
-  metricDataPoint: DataPoint[],
+  metricData: DataPoint[],
+  total: number,
   maxSize: number = 6,
 ): {
-  metricData: MetricDataType[];
+  metricData: DataPoint[];
   metricLegend: LegendDataType[];
-  total: number;
 } => {
-  let metricData: MetricDataType[] = [];
   const metricLegend: LegendDataType[] = [];
-  let total = 0;
 
-  if (!metricDataPoint || !metricDataPoint.length) {
-    return { metricData, metricLegend, total };
-  }
-  metricDataPoint.forEach((mDP) => {
-    const tmpMetricData: MetricDataType = { x: `${mDP.x}`, y: Number(mDP.y) };
-    // skip 'Others' label
-    if (tmpMetricData.x === 'Others') return;
-    total += tmpMetricData.y;
-    metricData.push(tmpMetricData);
-  });
-  metricData = _.sortBy(metricData, ['y']).reverse();
-  // collect the other data
-  const extraData = metricData.splice(maxSize, metricData.length);
-  // if there are any extra / other items, reduce it to a single entry
-  if (extraData.length) {
-    const otherData = extraData.reduce((prev, curr) => {
-      return { x: 'Others', y: prev.y + curr.y };
-    });
-    metricData.push(otherData);
+  // if there is no data, return the empty values
+  if (!metricData || !metricData.length) {
+    return { metricData, metricLegend };
   }
-  metricData = metricData.map((m) => {
-    metricLegend.push({ name: m.x });
-    return { x: m.x, y: (m.y / total) * 100 };
+  // get the percentage
+  metricData.forEach((m) => {
+    metricLegend.push({ name: `${m.x}` });
+    m.y = total ? (m.y / total) * 100 : 0;
```

Comment:
The value coming from prometheus is in bytes.
It needs to be converted into proper unit before using in chart. Not doing this will lead to discrepancy while showing **labels** . 
Try using `humanizeBinaryBytesWithoutB` for `m.y` and `total` (ensure that both the units be same)

Reply:
![Screenshot from 2019-08-06 17-04-27](https://user-images.githubusercontent.com/25664409/62536775-77f47900-b86c-11e9-8f9a-ad973c5c2754.png)

Reply:
> It needs to be converted into proper unit before using in chart. Not doing this will lead to discrepancy while showing labels
This should never happen, as we are showing the percentage. In the above graph, the data given has anomaly. The total value (the number shown in the middle of the chart) should always be higher than any of the individual value.
As we are calculating percentage, we don't have to do any conversions, but have to make sure that all the data arriving are of same unit...

Reply:
Okay I have checked , the mocked data was incorrect.